noether_simulator/src/noether_simulator.cpp

```diff
@@ -0,0 +1,399 @@
+/*
+ * Software License Agreement (Apache License)
+ *
+ * Copyright (c) 2016, Southwest Research Institute
+ *
+ * file noether_simulator.cpp
+ * All rights reserved.
+ * copyright Copyright (c) 2019, Southwest Research Institute
+ *
+ * License
+ * Software License Agreement (Apache License)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <noether_simulator/noether_simulator.h>
+
+#include <Eigen/Geometry>
+#include <vtkCylinderSource.h>
+#include <vtkDoubleArray.h>
+
+#include <vtkTransform.h>
+#include <vtkTransformPolyDataFilter.h>
+
+#include <vtkDelaunay3D.h>
+#include <vtkUnstructuredGrid.h>
+#include <vtkGeometryFilter.h>
+
+#include <vtkExtractPoints.h>
+#include <vtkSelectEnclosedPoints.h>
+#include <vtkModifiedBSPTree.h>
+#include <vtkLookupTable.h>
+
+namespace noether_simulator
+{
+  NoetherSimulator::NoetherSimulator()
+  {
+    process_base_rate_ = 5.0;
+    scalar_sigma_ = 3.0;
+    debug_on_ = false;
+    simulation_points_ = vtkSmartPointer<vtkPolyData>::New();
+  }
+
+  double NoetherSimulator::integral(double pt[3])
+  {
+
```

Comment:
why is the integral x^3/3 + z^3/3? Put some comments in here to help us understand what's going on